Library/Homebrew/unpack_strategy.rb

```diff
@@ -0,0 +1,241 @@
+class UnpackStrategy
+  def self.from(download:); end
+
+  def self.magic_number(path)
+    File.binread(path, 252)
+  end
+
+  def self.detect(path)
+    magic_number = magic_number(path)
+
+    strategies = [
+      JarUnpackStrategy,
```

Comment:
Can this be inferred somehow rather than hardcoded? If not, a class level constant feels a bit nicer.

Reply:
> a class level constant feels a bit nicer.

Not sure how a constant could replace this.

Reply:
A frozen constant array of classes rather than a variable.

Reply:
Ah, I thought you were talking about `JarUnpackStrategy`.